src/logic_utils/canonical_forms.h

```diff
@@ -0,0 +1,12 @@
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "truth_table_generator.cpp"
+#include "truth_table_generator.h"
+
+using namespace std;
+vector<Token> canonical_forms_sop(string expression);

```

Comment:
I suggest you have a seperate struct for Minterm, and one for Maxterm. and make two functions that return a `vector<Minterm>` and `vector<Maxterm>` . Then add a seperate function that coverts the `vector<Minterm>` into `vector<Tokens>` and so on. This way it will be more decoupled and easier to be used later by `generatePrimeImplicants` function.